paasta_tools/marathon_serviceinit.py

```diff
@@ -422,6 +432,18 @@ def status_mesos_tasks(service, instance, normal_instance_count):
         return "Error: talking to Mesos timed out. It may be overloaded."
 
 
+def get_marathon_dashboard_links(marathon_clients, system_paasta_config):
+    """Return a dict of marathon clients and their corresponding dashboard URLs"""
+    cluster = system_paasta_config.get_cluster()
+    try:
+        links = system_paasta_config.get_dashboard_links().get(cluster).get('Marathon RO')
+    except KeyError:
+        pass
+    if isinstance(links, list) and len(links) >= len(marathon_clients.current):
```

Comment:
Can this tolerate the situation where there is a smaller number of clients (2 shards) yet 3 dashboards defined?

Reply:
Yes, it can. [This assertion](https://github.com/Yelp/paasta/pull/1636/files#diff-f141232240ec690b401ca7d7aec03136R1180) tests such situation.